doc/man7/ossl-guide-quic-server-non-block.pod

```diff
@@ -0,0 +1,369 @@
+        fprintf(stderr, "cannot bind to %u\n", port);
+        BIO_closesocket(fd);
+        return -1;
+    }
+
+    /* Set port to nonblocking mode */
+    if (BIO_socket_nbio(fd, 1) <= 0) {
+        fprintf(stderr, "Unable to set port to nonblocking mode");
+        BIO_closesocket(fd);
+        return -1;
+    }
+
+To run the QUIC server, we create an B<SSL_LISTENER> to listen for incoming
+connections.  We provide it with the bound UDP port to then explicitly begin
+listening for new connections.
+
+    /* Create a new QUIC listener */
+    if ((listener = SSL_new_listener(ctx, 0)) == NULL)
+        goto err;
+
+    /* Provide the listener with our UDP socket. */
+    if (!SSL_set_fd(listener, fd))
+        goto err;
+
+    /* Set the listener mode to nonblocking, which is inherited by
+     * child objects.
+     */
+    if (!SSL_set_blocking_mode(listener, 0))
+        goto err;
+
+    /*
+     * Begin listening. Note that is not usually needed as SSL_accept_connection
+     * will implicitly start listening. It is only needed if a server wishes to
+     * ensure it has started to accept incoming connections but does not wish to
+     * actually call SSL_accept_connection yet.
+     */
+    if (!SSL_listen(listener))
+        goto err;
+
+=head2 Server loop
+
+The server now enters a "forever" loop, handling one client connection at a
+time.  Before each connection, we clear the OpenSSL error stack so that any
+error reports are related to just the new connection.
+
+    /* Pristine error stack for each new connection */
+    ERR_clear_error();
+
+We then wait until a connection is ready for reading.
+It uses the select function to wait until the socket is either readable
+or writable, depending on what the SSL connection requires.
+
+We then accept a new connection in which the handshake will have already
+occurred. However, since we are in nonblocking mode, L<SSL_accept_connection(3)>
+will return immediately. Therefore, we use a helper function to essentially
+block until a connection is established.
+
+    printf("Waiting for connection\n");
+    while ((conn = SSL_accept_connection(listener, 0)) == NULL) {
+        wait_for_activity(listener);
+    }
+    printf("Accepted new connection\n");
+
+The helper function wait_for_activity uses select() to block until the file
+descriptor belonging to the passed SSL object is readable. As mentioned earlier,
+a more real-world application would likely use this time to perform other tasks.
+
+    /* Initialize the fd_set structure */
+    FD_ZERO(&read_fd);
+    FD_ZERO(&write_fd);
+
+    /*
+     * Determine if we would like to write to the socket, read from it, or both.
+     */
+    if (SSL_net_write_desired(ssl))
+        FD_SET(sock, &write_fd);
+    if (SSL_net_read_desired(ssl))
+        FD_SET(sock, &read_fd);
+
+    /* Add the socket file descriptor to the fd_set */
+    FD_SET(sock, &read_fd);
+    FD_SET(sock, &write_fd);
```

Comment:
These two lines need a better explanation. They seem just to unconditionally duplicate what happens conditionally with SSL_net_{read,write}_desired above?

Or maybe they just make this code busy-wait is most cases?

Reply:
This is actually an error. Will remove it in the guide and also within the source code